src/Mod/Fem/feminout/importMshMesh.py

```diff
@@ -0,0 +1,249 @@
+# *   This file is part of the FreeCAD CAx development system.              *
+# *                                                                         *
+# *   This program is free software; you can redistribute it and/or modify  *
+# *   it under the terms of the GNU Lesser General Public License (LGPL)    *
+# *   as published by the Free Software Foundation; either version 2 of     *
+# *   the License, or (at your option) any later version.                   *
+# *   for detail see the LICENCE text file.                                 *
+# *                                                                         *
+# *   This program is distributed in the hope that it will be useful,       *
+# *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+__title__ = "Mesh import for .msh file format"
+__url__ = "https://www.freecad.org"
+
+## @package importMshMesh
+#  \ingroup FEM
+#  \brief FreeCAD Msh file reader for FEM workbench
+
+import os
+
+import FreeCAD
+from FreeCAD import Console
+from builtins import open as pyopen
+
+
+
+
+# ********* module specific methods *********
+
+def read(filename):
+    """read a FemMesh from a .msh mesh file and return the FemMesh
+    """
+
+    mesh_data = read_msh(filename)
+    from . import importToolsFem
+    return importToolsFem.make_femmesh(mesh_data)
+
+def read_msh(file_name):
+    """read .msh file """
+    # ATM only mesh reading is supported (no boundary conditions)
+
+    class elements():
+
+        tria3 = {}
+        tria6 = {}
+        quad4 = {}
+        quad8 = {}
+        tetra4 = {}
+        tetra10 = {}
+        hexa8 = {}
+        hexa20 = {}
+        penta6 = {}
+        penta15 = {}
+        seg2 = {}
+        seg3 = {}
+    nodes = {}
+    contigous_numbered = False
+
+    f = pyopen(file_name, "r")
+    line = "\n"
+    total_num_nodes = 0
+    node_ind = 1
+    node_tag_array = []
+    tag_ind = 0
+    node_coord = False
+    num_element = 0
+    number_of_nodes = 0
+    read_elements = False
+    read_node = False
+    elm_category = []
+    point_elem = False
+    number_of_nodes = 0
+    error_seg3 = False
+    error_not_supported_elemtype = False
+    while line != "":
+        line = f.readline()
```

Comment:
It's usually more idiomatic to `for line in f.readlines()`

Reply:
I thought of using `for` loop in this but seeing that `while` loop is being used for other file types, I wanted to keep the code consistent.